translate-pixiv-tags.user.js

```diff
@@ -761,7 +779,14 @@ function addDanbooruTags ($target, tags, options = {}) {
 async function translateArtistByURL (element, profileUrl, options) {
     if (!profileUrl) return;
 
-    const artistUrls = await queueNetworkRequestMemoized("url", normalizeProfileURL(profileUrl));
+    const promiseArray = [];
+    promiseArray.push(queueNetworkRequestMemoized("url", normalizeProfileURL(profileUrl)));
+    const translatedUrl = translateProfileURL(profileUrl);
+    if (translatedUrl !== profileUrl) {
+        promiseArray.push(queueNetworkRequestMemoized("url", normalizeProfileURL(translatedUrl)));
+    }
+    const data = await Promise.all(promiseArray);
+    const artistUrls = [].concat([], ...data);
```

Comment:
It's not to copy the array but to flatten two arrays.

The data returned by the Promise all looks like one of the following...

`[[results1]]` or `[[results1],[results2]]`

...which are two dimensional arrays, whereas we need it to look like the following...

`[results1 + results2]`

... which is a one dimensional array. Normally that could be accomplished by using the `flat` function, but as I recall that was an issue with Firefox since it didn't have that function. However I just tested it, and it seems that at least version 69 of Firefox does have it so I'll change it to that then.